utilities/find_optimal_probability_threshold/predictor.py

```diff
@@ -19,12 +19,12 @@ def preprocess_text(self, text):
 						  strip_punctuation, strip_multiple_whitespaces]
 		return unidecode(' '.join(preprocess_string(text, custom_filters)))
 
-	def predict_ror_id(self, affiliation, confidence):
+	def predict_ror_id(self, affiliation, min_probability):
 		affiliation = self.preprocess_text(affiliation)
-		predicted_label = self.classifier.predict(affiliation, k=1)
-		label, ratio = predicted_label[0][0], predicted_label[1][0]
-		if ratio >= confidence:
+		predicted_label = self.classifier.predict(affiliation, k=1, threshold=min_probability)
+		if predicted_label and predicted_label[0] and predicted_label[1]:
+			label, ratio = predicted_label[0][0], predicted_label[1][0]
 			label = re.sub('__label__','', label)
-			return [label, ratio]
+			return label, ratio
 		else:
-			return None, None
\ No newline at end of file
+			return None
```

Comment:
Updated in https://github.com/ror-community/affiliation-matching-experimental/pull/14/commits/8b60e937443debe4458853b69124f2ccb76af708. Let me know if this is what you had in mine for simplifying.

Reply:
Now I am not sure whether your client code is still correct. After doing:

```
fasttext_prediction = PREDICTOR.predict_ror_id(affiliation, min_fasttext_probability)
```

fasttext_prediction is a tuple with 2 elements. So when you do `if fasttext_prediction:` it will always be true.

What I meant in the original comment is that the change you applied to the values returned by predict_ror_id lets you unpack immediately:

```
predicted_ror_id, prediction_confidence = PREDICTOR.predict_ror_id(affiliation, min_fasttext_probability)
```

and then check only predicted_ror_id in the if.

Thanks to this, it is immediately clear for the reader what exactly is returned by predict_ror_id. Without it, you either unpack the values a few lines later in the code (and by that time the reader will forget where fasttext_prediction variable came from), or you have to use confusing indexes (and to understand what fasttext_prediction[0] is, as a reader I have to consult another file). So I believe unpacking immediately makes the code clearer.

Similar changes should be applied in concurrence_check script as well.